src/test/java/study/StringAddCalculator.java

```diff
@@ -0,0 +1,47 @@
+package study;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+public class StringAddCalculator {
+    static int splitAndSum(String text){
+        int sum = 0;
+        if (text==null || text.isEmpty()){
+            return 0;
+        }
+        if (text.length()==1){
+            int number = Integer.parseInt(text);
+            return number;
+        }
+        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(text);
+        if (m.find()){
+            String customDelimiter = m.group(1);
+            String[] tokens= m.group(2).split(customDelimiter);
+            sum = sumAll(tokens);
+            if(sum > 0) return sum;
+
+        }
+        else{
+            String[] numbers = text.split(",|:");
+            sum = sumAll(numbers);
+
+            if(sum > 0) return sum;
+        }
+
+        return -1;
+    }
+    public static int sumAll (String[] s){
+        int total=0;
```

Comment:
```suggestion
        int total = 0;
```
전체적으로 (띄어쓰기와 같은)컨벤션이 안 맞춰져있는 것 같은데요~
인텔리제이를 사용하신다면 `ctrl` + `alt` + `L` (or `opt` + `cmd` + `L`)을 사용한 포맷팅을 습관화하시거나, 오토 포맷팅 설정을 추천드립니다ㅎㅎ